src/BlazorDatasheet.Core/Commands/Data/UnMergeCellsCommand.cs

```diff
@@ -0,0 +1,70 @@
+﻿using BlazorDatasheet.Core.Data;
+using BlazorDatasheet.Core.Data.Cells;
+using BlazorDatasheet.Core.Formats;
+using BlazorDatasheet.DataStructures.Geometry;
+
+namespace BlazorDatasheet.Core.Commands.Data;
+
+public class UnMergeCellsCommand : IUndoableCommand
+{
+    private readonly List<IRegion> _unMergesPerformed = new();
+    private CellStoreRestoreData _restoreData = null!;
+    private readonly IRegion _region;
+
+    /// <summary>
+    /// Command that unmerges the cells in the range given.
+    /// Note that the value in the merged range will return to the top left cell, while the
+    ///   other cells will be blank.
+    /// </summary>
+    /// <param name="range">The range in which to merge. </param>
+    public UnMergeCellsCommand(IRegion region)
+    {
+        _region = region.Clone();
+    }
+
+    public bool CanExecute(Sheet sheet)
+    {
+        var existingMerges = sheet.Cells.GetMerges(_region).ToList();
+        if (existingMerges.All(x => _region.Contains(x)))
+            return true;
+        return false;
+    }
+
+    public bool Execute(Sheet sheet)
+    {
+        sheet.BatchUpdates();
+        _unMergesPerformed.Clear();
+
+        var region = _region;
+
+        // Clear all the cells that are not the top-left posn of merge and store their values for undo
+        var regionsToClear = region
+            .Break(region.TopLeft)
+            .ToList();
+
+        //Store the values that are being cleared for undo
+        _restoreData = sheet.Cells.ClearCellsImpl(regionsToClear);
```

Comment:
I don't believe we will need to clear any cells when un-merging as there shouldn't be any values set under the merge so we could get rid of this and not worry about storing this as undo data.

Reply:
If the `_restoreData` property is not set, a user will run into an error when trying to undo an unmerge command. Is there another method you would recommend using in place of the `ClearCellsImpl()` method here to ensure we still store the merged state?

Reply:
Is the error because the command is trying to restore the data still (line 65 of [UnMergeCommand.cs](https://github.com/anmcgrath/BlazorDatasheet/blob/0daecd55dd8e474a60ec868b827ab46baebaa68e/src/BlazorDatasheet.Core/Commands/Data/UnMergeCellsCommand.cs)?

If it does need to store the data it would be good to have a test that shows it.

Reply:
Yes, you were correct. We can do without saving any of the cell data and just re-merge the sections during the undo. Thank you! I will push these changes.